server/src/test/java/org/elasticsearch/rest/RestResponseTests.java

```diff
@@ -85,19 +85,6 @@ public void testDetailedExceptionMessage() throws Exception {
             {"type":"file_not_found_exception","reason":"/foo/bar"}"""));
     }
 
-    public void testNonElasticsearchExceptionIsNotShownAsSimpleMessage() throws Exception {
```

Comment:
Why did we remove this test altogether? Do we have other tests covering what happens if non-elasticsearch exception is thrown? If the expected outcome is the same, perhaps we can refactor the above tests to demonstrate the error is consistent between throwing ES and non-ES exception.

Reply:
This tests that the error message is just the string "No ElasticsearchException found", which is no longer the behaviour of this class. Checking the error messages when detailed is off is now covered by `testGenerateFailureToXContentWithNoDetails`, rendering this test superfluous.